src/python/pants/backend/jvm/tasks/jvm_task.py

```diff
@@ -75,15 +75,17 @@ def classpath(self, targets, classpath_prefix=None, classpath_product=None, excl
 
   def preferred_jvm_distribution_for_targets(self, targets, jdk=False, strict=False):
     """Find the preferred jvm distribution for running code from the given targets."""
+    return self.preferred_jvm_distribution(
+      self.runtime_platforms_for_targets(targets),
+      jdk=jdk, strict=strict)
+
+  def runtime_platforms_for_targets(self, targets):
     target_platforms = [self._runtime_platform_for_target(target) for target in targets if
       isinstance(target, JvmTarget)]
     if not target_platforms:
-      platforms = [JvmPlatform.default_runtime_platform]
+      return [JvmPlatform.default_runtime_platform]
     else:
-      platforms = target_platforms
-    return self.preferred_jvm_distribution(
-      platforms,
-      jdk=jdk, strict=strict)
+      return target_platforms
```

Comment:
This could possibly be converted into:
```python
      return target_platforms or [JvmPlatform.default_runtime_platform]
```
Your change allowed me to unlock this!

Reply:
I think the `if` here makes the semantics clearer, but it could work that way too! I'd prefer to leave it.

Reply:
Ok, totally great!